src/dp_abstraction/aca_dataplane_ovs.cpp

```diff
@@ -121,8 +121,11 @@ int ACA_Dataplane_OVS::update_port_state_workitem(const PortState current_PortSt
     generated_port_name = aca_get_port_name(current_PortConfiguration.id());
 
     try {
-      // TODO: add support for more than one fixed_ips in the future
-      assert(current_PortConfiguration.fixed_ips_size() == 1);
+      if (current_PortConfiguration.fixed_ips_size() <= 0) {
+        ACA_LOG_ERROR("PortConfiguration.fixed_ips_size: %d.\n",
+                      current_PortConfiguration.fixed_ips_size());
+        throw std::invalid_argument("PortConfiguration.fixed_ips_size is less than zero");
+      }
       virtual_ip_address = current_PortConfiguration.fixed_ips(0).ip_address();
```

Comment:
Although we allow more than fixed IPs, we process the first one only as the primary IP.  As a next step, it would be awesome to start thinking about storing the secondary IPs so that when customers add IPs in the VM (manually as limited by current networking stack), Control Plane has the knowledge to grant or reject customers' traffic, depending on whether customers configure the VM in the right way.

Reply:
got it. The current port creation code doesn't consume the virtual IP address(s). It would be consumed in security group code to allow traffic come out from the VM with certain virtual IPs and virtual mac (it also should look at the allow_address_pairs in PortConfiguration).

Reply:
Let me rephrase the question: if an interface has more than 1 ports, the current ACA would only process and store the first ip, and other ips are ignored. We will need to handle other ips too.